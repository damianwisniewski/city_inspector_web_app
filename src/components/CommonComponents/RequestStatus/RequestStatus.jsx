import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

// Components
import Loader from '../Loader/Loader'
import Status from '../Status/Status'

const RequestStatus = ({ id, requestState, children, errorMessage, direction }) => (
	<Fragment>
		{requestState === 'pending' && <Loader id={`${id}-loader`} />}
		{requestState === 'failed' && (
			<Status id={`${id}-status`} position={direction} type='error' message={errorMessage} />
		)}
		{requestState === 'succeeded' && children}
		{requestState === 'initial' && children}
	</Fragment>
)

RequestStatus.defaultProps = {
	direction: 'horizontal',
}

RequestStatus.propTypes = {
	requestState: PropTypes.string.isRequired,
	children: PropTypes.node.isRequired,
	errorMessage: PropTypes.string.isRequired,
	direction: PropTypes.string,
}

export default RequestStatus
