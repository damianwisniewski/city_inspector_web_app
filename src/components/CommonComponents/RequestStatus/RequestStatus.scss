@import '../../../assets/styleHelpers/index';

.StatusScreen {
	display: flex;
	align-items: center;
	justify-content: center;
	width: 100%;
	height: 100%;

	&--fullview {
		height: 100vh;
	}

	&--transparent {
		background-color: transparent;
	}

	&--blue {
		background-color: $primary_light_blue;
	}
}
