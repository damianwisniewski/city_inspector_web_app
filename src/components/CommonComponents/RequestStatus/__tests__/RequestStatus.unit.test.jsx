import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import RequestStatus from '../RequestStatus'

configure({ adapter: new Adapter() })

describe('<RequestStatus />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let RequestStatusElement
	const id = 'FakeID'
	const children = (
		<div>
			<p>Test</p>
		</div>
	)
	const errorMessage = 'Fake error message'

	test('should render correctly RequestStatus with requestState pending', () => {
		RequestStatusElement = shallow(
			<RequestStatus id={id} requestState='pending' errorMessage={errorMessage}>
				{children}
			</RequestStatus>,
		)
		expect(toJson(RequestStatusElement)).toMatchSnapshot()
	})

	test('should render correctly RequestStatus with requestState failed', () => {
		RequestStatusElement = shallow(
			<RequestStatus id={id} requestState='failed' errorMessage={errorMessage}>
				{children}
			</RequestStatus>,
		)
		expect(toJson(RequestStatusElement)).toMatchSnapshot()
	})

	test('should render correctly RequestStatus with requestState succeeded', () => {
		RequestStatusElement = shallow(
			<RequestStatus id={id} requestState='succeeded' errorMessage={errorMessage}>
				{children}
			</RequestStatus>,
		)
		expect(toJson(RequestStatusElement)).toMatchSnapshot()
	})

	test('should render correctly RequestStatus with direction horizontal', () => {
		RequestStatusElement = shallow(
			<RequestStatus
				id={id}
				requestState='initial'
				direction='horizontal'
				errorMessage={errorMessage}
			>
				{children}
			</RequestStatus>,
		)
		expect(toJson(RequestStatusElement)).toMatchSnapshot()
	})

	test('should render correctly RequestStatus with direction vertical', () => {
		RequestStatusElement = shallow(
			<RequestStatus
				id={id}
				requestState='initial'
				direction='vertical'
				errorMessage={errorMessage}
			>
				{children}
			</RequestStatus>,
		)
		expect(toJson(RequestStatusElement)).toMatchSnapshot()
	})
})
