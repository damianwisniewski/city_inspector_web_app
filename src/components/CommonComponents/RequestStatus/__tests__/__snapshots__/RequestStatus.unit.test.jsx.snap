// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestStatus /> should render correctly RequestStatus with direction horizontal 1`] = `
<Fragment>
  <div>
    <p>
      Test
    </p>
  </div>
</Fragment>
`;

exports[`<RequestStatus /> should render correctly RequestStatus with direction vertical 1`] = `
<Fragment>
  <div>
    <p>
      Test
    </p>
  </div>
</Fragment>
`;

exports[`<RequestStatus /> should render correctly RequestStatus with requestState failed 1`] = `
<Fragment>
  <Status
    id="FakeID-status"
    message="Fake error message"
    position="horizontal"
    type="error"
  />
</Fragment>
`;

exports[`<RequestStatus /> should render correctly RequestStatus with requestState pending 1`] = `
<Fragment>
  <Loader
    id="FakeID-loader"
    size="medium"
  />
</Fragment>
`;

exports[`<RequestStatus /> should render correctly RequestStatus with requestState succeeded 1`] = `
<Fragment>
  <div>
    <p>
      Test
    </p>
  </div>
</Fragment>
`;
